rasa/graph_components/validators/config_validator.py

```diff
@@ -0,0 +1,532 @@
+        TODO: this also works the other way round, we can warn/raise if a
+            component won't be trained instead of creating a placeholder model
+
+        Args:
+            training_data: training data
+        """
+        if (
+            training_data.response_examples
+            and ResponseSelectorGraphComponent not in self._component_types
+        ):
+            rasa.shared.utils.io.raise_warning(
+                f"You have defined training data with examples for training a response "
+                f"selector, but your NLU pipeline does not include a response selector "
+                f"component. To train a model on your response selector data, add a "
+                f"'{ResponseSelectorGraphComponent.__name__}' to your pipeline."
+            )
+
+        if training_data.entity_examples and self._component_types.isdisjoint(
+            TRAINABLE_EXTRACTORS
+        ):
+            rasa.shared.utils.io.raise_warning(
+                "You have defined training data consisting of entity examples, but "
+                "your NLU pipeline does not include an entity extractor trained on "
+                "your training data. To extract non-pretrained entities, add one of "
+                f"{types_to_str(TRAINABLE_EXTRACTORS)} to your pipeline."
+            )
+
+        if training_data.entity_examples and self._component_types.isdisjoint(
+            {DIETClassifierGraphComponent, CRFEntityExtractorGraphComponent}
+        ):
+            if training_data.entity_roles_groups_used():
+                rasa.shared.utils.io.raise_warning(
+                    f"You have defined training data with entities that "
+                    f"have roles/groups, but your NLU pipeline does not "
+                    f"include a '{DIETClassifierGraphComponent.__name__}' "
+                    f"or a '{CRFEntityExtractorGraphComponent.__name__}'. "
+                    f"To train entities that have roles/groups, "
+                    f"add either '{DIETClassifierGraphComponent.__name__}' "
+                    f"or '{CRFEntityExtractorGraphComponent.__name__}' to your "
+                    f"pipeline."
+                )
+
+        if training_data.regex_features and self._component_types.isdisjoint(
+            [RegexFeaturizerGraphComponent, RegexEntityExtractorGraphComponent],
+        ):
+            rasa.shared.utils.io.raise_warning(
+                f"You have defined training data with regexes, but "
+                f"your NLU pipeline does not include a 'RegexFeaturizer' or a "
+                f"'RegexEntityExtractor'. To use regexes, include either a "
+                f"'{RegexFeaturizerGraphComponent.__name__}' or a "
+                f"'{RegexEntityExtractorGraphComponent.__name__}' in your pipeline."
+            )
+
+        # NOTE: bugs
+        # 1. removed the check that complains if neither "CRFEntityExtractor",
+        #   nor "DIETClassifier" are contained in case a lookup table is given...
```

Comment:
Does DIET ignore all features by the `RegexFeaturizer` or only the lookup ones?

Reply:
DIET is not using the `training_data.lookup_table` anywhere (otherwise, I'm making some mistake here - currently refactoring that data pipeline and testing it 😅 ). Features added to the messages via the `RegexFeaturizer` would be used - iff it exists. So if a DIETClassifier is in the pipeline, the lookup table might still be un-used.

Reply:
Thanks for the explanation! So the actual warning should be: "You need `RegexFeaturizer + patterns enabled and a trainable intent classifier" to make use of lookup tables?

Reply:
Oh, ok, now I see what you mean / why these checks where like this. 

The check should be " RegexFeaturizer + something consuming those features " and in NLU the CRF entity extractor with "patterns" and the DIETClassifier are really the only consumers 😅 . SklearnIntentClassifier could/should also be but that only makes use of dense sentence features for some reason. In Core, TED should pick those features up as well... but I guess we ignore this for now with v1.  I'll just add these checks back in then.

Reply:
... btw, I just mixed up regexes and lookup tables --  the same applies (should apply) to the regex part of the training data 🙈 . If training data has regexes, and a regex extractor (even with `use_regexes == False`) and we have no `DIETClassifier` then we will not get a warning at the moment. I'll revert to the old logic for now, put in some `TODO`s, and (commented out) tests to illustrate what's missing.

Think we should keep this in mind when revisiting the "required components". We could add something like "possible/providing/<insert better name here> components". Something to reflect that e.g. SklearnIntentClassifier does not only need dense featurizers but won't use the sparse features produced by the RegexFeaturizer.

Reply:
Can we skip the Todos and create issues instead? We'll never do them otherwise 😄

Reply:
👍

Reply:
https://github.com/RasaHQ/rasa/issues/9705